tools/advisory-parser/src/models.rs

```diff
@@ -0,0 +1,182 @@
+                                    "GHSA contains non-ASCII characters",
+                                ));
+                            }
+                            // GHSA format: GHSA-xxxx-xxxx-xxxx using base32 chars (excluding 0,1,a,e,i,l,o,s,u)
+                            let re =
+                                Regex::new(r#"^GHSA(-[23456789cfghjmpqrvwx]{4}){3}$"#).unwrap();
+                            if !re.is_match(&value) {
+                                return Err(de::Error::custom(
+                                    "Invalid GHSA format: must match GHSA-xxxx-xxxx-xxxx",
+                                ));
+                            }
+                            ghsa = Some(value);
+                        }
+                        "severity" => severity = Some(map.next_value()?),
+                        "description" => description = Some(map.next_value()?),
+                        "products" => products = Some(map.next_value()?),
+                        _ => {
+                            return Err(de::Error::unknown_field(
+                                &key,
+                                &[
+                                    "id",
+                                    "title",
+                                    "cve",
+                                    "ghsa",
+                                    "severity",
+                                    "description",
+                                    "products",
+                                ],
+                            ))
+                        }
+                    }
+                }
+
+                // Require at least one security identifier
+                if cve.is_none() && ghsa.is_none() {
+                    return Err(de::Error::custom(
+                        "At least one of 'cve' or 'ghsa' must be present",
+                    ));
+                }
+
+                Ok(AdvisoryInfo {
+                    id: id.ok_or_else(|| de::Error::missing_field("id"))?,
+                    title: title.ok_or_else(|| de::Error::missing_field("title"))?,
+                    cve,
+                    ghsa,
+                    severity: severity.ok_or_else(|| de::Error::missing_field("severity"))?,
+                    description: description
+                        .ok_or_else(|| de::Error::missing_field("description"))?,
+                    products: products.ok_or_else(|| de::Error::missing_field("products"))?,
+                })
+            }
+        }
+
+        deserializer.deserialize_struct(
+            "AdvisoryInfo",
+            &[
+                "id",
+                "title",
+                "cve",
+                "ghsa",
+                "severity",
+                "description",
+                "products",
+            ],
+            AdvisoryInfoVisitor,
+        )
+    }
+}
+
+/// Package metadata affected by the advisory.
+#[derive(Deserialize)]
+pub struct Product {
```

Comment:
You can do this instead of renaming each individual field

```suggestion
#[derive(Deserialize)]
#[serde(rename_all = "kebab-case")]
pub struct Product {
```